### DISCUSSION TITLE
Issues when using the OSMAndMap Pro App on iPhone-13 and iPad(7) devices

### DISCUSSION THREAD

**Discussion by Rmacca** [osmandapp/OsmAnd#19895] [Q&A]
**(This is a copy of an email I sent to OSM Support and they re-directed me to here so I have just posted the email that I had sent to them, so hopefully it displays as I intended with attached Images relating to my issues)**

I have been using the “OsmandMaps (Pro) version” App for a few years now and I have recently started to notice that I am having some issues with the App on my iPad and to some extent on my iPhone as well. I am running the latest IOS software on both ios devices. My 10’’ iPad is the 7th generation (built-in GPS receiver) iPad, and my iPhone is the iPhone-13.
The issues I am having are as follows:

- At random times, I have had the OSMandMap App just shut down for no reason during my trips while I’ve been using the App to navigate.  I then must restart the App again to ensure that the recording of my current TRIP Tracklog continues, otherwise I have found that it does not record my current Tracklog.  This has happened on both my Apple iPad and the iPhone 13 devices.

**Re: TRACKS section:**

- When importing a planned Trip GPX file that I have created that only contains “WAYPOINTS” into the **OSMandMap Pro App**, it displays in the “Tracks” area **(My Places > Tracks)** as per normal, but I’ve started to notice that after some time that even though the App indicates that the GPX file is visible on the map, I can no longer see any of my **Waypoints (favourites)** on the Map.  When I look the “My Tracks” section where it lists all your folders & GPX files, it displays that the 2024 (Jun) WAYPOINTS – Exploring North Simpson Desert (Plenty Lakes) Trip GPX file contains 405 waypoints(favourites) – see below:

![1](https://github.com/osmandapp/OsmAnd/assets/170098817/dfef3c65-b4a6-46c6-ba5b-1b5d32190601)

But when I click on the gpx file to open it where I can then modify/edit it via clicking on the "**Actions**” button, I get the following:
![2](https://github.com/osmandapp/OsmAnd/assets/170098817/43b6d62d-88f3-4af5-b399-747bd4884617)

Normally, on the left-hand side, it would list the **405 Waypoints** contained within the GPX File. As you can see there is nothing there. This is very frustrating as the only way I know to fix this is to “**DELETE**” the GPX file and then “**RELOAD**” it again into the OSM App and then do not try to edit any of the waypoints within the GPX file. 

**Below is an image of what should be displayed when I click on the GPX file to open it so I can edit/modify the waypoints:**
![3](https://github.com/osmandapp/OsmAnd/assets/170098817/7f14dbd8-7cf6-4ca0-b217-7ed74f9b1529)

**- PLEASE NOTE:** I have also had the **EXACT** same scenario (Problem) that I have described above with **WAYPOINITS** happen to me but this time it’s a GPX File that only contains “**TRACKS**” and NO waypoints. The same scenario applies where the GPX file is showing that it is visible on the map, but I do not see any of the Tracks being displayed on the Map. I have found that if I "Re-load" the GPX file into OSM app again and DO NOT then go into the file to modify anything, then it is likely to work as it is intended, but if I need to edit the GPX file and change the colour of a Track, that this is when I “could” lose the ability to see the Tracks on the Map. This is VERY FRUSTRATING… **(SEE IMAGE BELOW) - Even though OSM is indicating that the Track is visible on the MAP, you can see that NO TRACKS are shown on the MAP**

![4](https://github.com/osmandapp/OsmAnd/assets/170098817/d646c667-47a1-4334-91b7-3fae9bd64d11)

**Re: FAVOURITES section:**

- I have tried importing my list of GPX Waypoints (**Favourites**) into the “Favourites” section to help prevent the waypoints from disappearing on the map.  This has worked to a point. Once I have imported my **Waypoints** into the **Favourites** area, I then want to "**Rename**" the Folder &/or Modify/Edit the **Favourites** Colour &/or Icons displayed.  **This is where I run into problems. When I click on the "SAVE" button to save the changes, the OSMandMap App immediately SHUTS DOWN completely WITHOUT SAVING any of the changes that I've made.  I have since worked out that I can ONLY CHANGE ICON COLOURS and then successfully SAVE the changes but if I do anything else, the App SHUTS DOWN immediately.  Again - very frustrating...**

**PLEASE NOTE:**   **The ISSUES that I am experiencing are mainly on my Apple iPad. I can do ALL that I described above on my iPhone successfully, BUT I am still finding that the OSMandMap App shuts down randomly (ie: without me touching or doing anything) on my phone. Again this is VERY FRUSTRATING when using my iPhone for navigational purposes...**

**Is there LIMITS within the OSMandMAP Pro App on the following:**

- Is there a limit to the **NUMBER** of WAYPOINTS (Favourites) that I can IMPORT? (Can I import ANY NUMBER of Waypoints?)
- Is there a limit to the **NUMBER** of TRACKS that I can IMPORT? (Can I import ANY NUMBER of TRACKS?)
- Is there a limit to the **SIZE** of the TRACKS that I can IMPORT? (Can I import ANY SIZE of TRACKS? - eg: track that contains >100,000 points)

**Can you please advise me if there is any software Bugs that I am not aware of or if I am doing something wrong, then could you please let me accordingly?**

**Comment by Rmacca** [comment]
Has anyone experienced any of the issues that I've mentioned above?

**Reply by vshcherb** [reply]
There is no defined limit though each device has limits for maximum processing points. We aim to support display of tracks with 100K-1M points in total and Favorites 1K - 10K points

**Reply by Rmacca** [reply]
**OK, thanks for that, but any idea or is there past records on the following:**
- Why can't I bulk change My Favourites "Icons" &/or Favourites Folder name? (Every time I try this, it causes OSM to shutdown) 
- Why does OSMAnd randomly shutdown on my ios devices when using it to navigate?

**Reply by vshcherb** [reply]
- If you have crashes, you can send them and mark somehow ```crashes from Rmacca``` so we can check the crashes and fix them. For bulk editing OsmAnd is not the right tool I guess, you can always find a PC tool to edit GPX with lots of point, so you can later use it

**Reply by Rmacca** [reply]
Thank you for your reply...
The problem is that I develop my Tracklogs and Waypoints (+icon type) within Garmin's Basecamp. I then upload this into OSM but whenever I've done this the waypoint icons all default to "Blue Star" (currently). I have not been able to work out how to change this Default Icon when I import a GPX file.
Doing a "Bulk Change" to the Waypoints (Favourites) within OSM is the only way I can see to make the changes I want to do (i.e.: relevant ICON and icon COLOUR). 
Would be nice if one could do these Bulk changes from within the OSM app and change the waypoint icon to reflect visually what the waypoint (Favourite) relates to (i.e.: Building; Car; Mountain, etc).
